From 19356084c2735928adfb1f9c8f0e634242aa7d47 Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Thu, 9 Apr 2015 18:23:01 +0800
Subject: [PATCH] MLK-10660-3 video: mxc ipuv3 fb: Correct prefetch flag if no
 PRE is available

For those SoCs which have no PRE engines embedded, users may try to call
the ioctrl MXCFB_SET_PREFETCH to enable the PRE.  The ioctrl sets the flag
mxcfbi->prefetch to be true to go for PRE allocation.  It is doomed to fail
and the users will have no chance to set the flag back to false.  As a result,
the framebuffer will not be responsive to the users' set_par ioctls.  Let's
set the current prefetch flag back to the prefetch flag in that case to avoid
the situation.

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
---
 drivers/video/mxc/mxc_ipuv3_fb.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/video/mxc/mxc_ipuv3_fb.c b/drivers/video/mxc/mxc_ipuv3_fb.c
index 23b04e6..aa13b02 100644
--- a/drivers/video/mxc/mxc_ipuv3_fb.c
+++ b/drivers/video/mxc/mxc_ipuv3_fb.c
@@ -566,6 +566,7 @@ static int _setup_disp_channel2(struct fb_info *fbi)
 			if (mxc_fbi->pre_num < 0) {
 				dev_err(fbi->device,
 					"failed to alloc PRE\n");
+				mxc_fbi->prefetch = mxc_fbi->cur_prefetch;
 				return mxc_fbi->pre_num;
 			}
 		}
-- 
1.8.0

